pip/req/req_cache.py

```diff
@@ -0,0 +1,217 @@
+from __future__ import absolute_import
+
+from collections import defaultdict
+import logging
+import os.path
+import shutil
+import tempfile
+
+from pip.download import url_to_path
+from pip.utils import display_path, ensure_dir, rmtree, _make_build_dir
+
+
+logger = logging.getLogger(__name__)
+
+
+class RequirementCache(object):
+    """A cache of requirements.
+
+    :attr delete: May be set to False to disable deleting the cache.
```

Comment:
The first time I read this I thought "deleting the cache" meant an in-memory object; reading the docstring for `__init__` I realised that it actually refers to the on-disk cache artifacts 

Maybe "deleting the cached artifacts" or "deleting the cached files" would be clearer here, but I think this is a minor issue.